### PR TITLE
[@types/vast-client] Updated interfaces for breaking changes in version 3.0

### PR DESCRIPTION
**Pull request by ngehlert**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/dailymotion/vast-client-js/blob/master/docs/api/3.0-migration.md

Linked to this issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/50693

Currently I'm still using this library actively, if you want to and need help with maintaining this I can add myself to the "Definitions by" list @jgainfort

### CODE REVIEW

**Comment by ngehlert** [comment]
added couple of tests. not for all cases since the structure was not available for everything. is this the way you imagined the tests to be?

File: types/vast-client/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for vast-client 2.1
+// Type definitions for vast-client 3.0
 // Project: https://github.com/dailymotion/vast-client-js#readme
 // Definitions by: John G. Gainfort Jr. <https://github.com/jgainfort>, Sara Nordmyr da Cunha <https://github.com/kobawan>
```
Comment (jgainfort) [inline types/vast-client/index.d.ts:3]: @ngehlert add yourself to this list :)

**Review by jgainfort — CHANGES_REQUESTED** [review]
This is a great start IMO but would be nice to finish the rest of the migration doc. For example:

Updates to track function signature.
VastTracker.track(event, once) -> VastTracker.track(event, {macros, once})
https://github.com/dailymotion/vast-client-js/blob/master/docs/api/3.0-migration.md#version-300-alpha8

Updates to fetchVast and parseVastXml function signature.
Payload changes, parameter name changes.
https://github.com/dailymotion/vast-client-js/blob/master/docs/api/3.0-migration.md#version-300-alpha8

Im cool with approving for now if you need to get this through. Thanks for the PR!

**Comment by martin-badin** [comment]
@ngehlert Please move the old version of this library to separate folder `v2`.

**Comment by ngehlert** [comment]
@martin-badin  so vast-client-v2? Whats the purpose of the separate folder?

@jgainfort thanks for your review. Must have missed those couple of lines. Will update my PR

**Comment by ngehlert** [comment]
changes are pushed. still don't quite know what am I supposed to do with the separate folder

**Review by jgainfort — APPROVED** [review]

**Comment by jgainfort** [comment]
@ngehlert Looks good.

I think @martin-badin means is to create a v2 folder under vast-client folder and put in the old definition and test files into it.

**Comment by ngehlert** [comment]
ok but then still, what would be the purpose of those files? as they are already available via git and released as a separate package

**Comment by jgainfort** [comment]
🤷

**Comment by martin-badin** [comment]
@ngehlert Maybe exists someone who is using an old version of this package and he wants to use the correct version of this type.

**Comment by ngehlert** [comment]
@martin-badin but wouldn't it make more sense for him to use the old library version 2.1 then? I mean following this logic we would've need a separate directory for every version that introduces breaking changes.

**Comment by martin-badin** [comment]
@ngehlert we should create a folder only for major version. You can see jquery for example but I think a lot of packages using these folder to separate major versions.

**Comment by ngehlert** [comment]
@martin-badin @jgainfort ah ok, I got it. I think it is to keep old typings, in case if there are bugs or updates in older version lines. Because otherwise we couldn't update them anymore because there is no way (in this repo) to release version 2.1.5. for a file if you are already on version 3.0
I added a new commit and added all the current files to a v2 directory.

Would be great if you could quickly verify those changes again :) 
Thanks

**Review by martin-badin — APPROVED** [review]
Thanks. Looks good to my 🙂💪👌

**Comment by ngehlert** [comment]
hm currently still checking why CI is failing

**Comment by ngehlert** [comment]
ah ok, you have to adapt baseUrl and typeRoots. fixed it. CI should now run through

**Review by martin-badin — APPROVED** [review]

**Pull request merged**